File: Chapter06/Java/CodeInterface/app/src/main/java/com/androidbook/codeinterface/MainActivity.java
package com.androidbook.codeinterface;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private LinearLayout mNameContainer;
    private LinearLayout mAddressContainer;
    private LinearLayout mParentContainer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        createNameContainer();
        createAddressContainer();
        createParentContainer();
        setContentView(mParentContainer);
    }

    private void createNameContainer() {
        mNameContainer = new LinearLayout(this);
        mNameContainer.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.WRAP_CONTENT));
        mNameContainer.setOrientation(LinearLayout.HORIZONTAL);

        TextView nameLbl = new TextView(this);
        nameLbl.setText("Name:");

        TextView nameValue = new TextView(this);
        nameValue.setText("John Doe");

        mNameContainer.addView(nameLbl);
        mNameContainer.addView(nameValue);
    }

    private void createAddressContainer() {
        mAddressContainer = new LinearLayout(this);
        mAddressContainer.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.WRAP_CONTENT));
        mAddressContainer.setOrientation(LinearLayout.HORIZONTAL);

        TextView addrLbl = new TextView(this);
        addrLbl.setText("Address:");

        TextView addrValue = new TextView(this);
        addrValue.setText("911 Hollywood Blvd");

        mAddressContainer.addView(addrLbl);
        mAddressContainer.addView(addrValue);
    }

    private void createParentContainer() {
        mParentContainer = new LinearLayout(this);
        mParentContainer.setLayoutParams(new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.MATCH_PARENT));
        mParentContainer.setOrientation(LinearLayout.VERTICAL);

        mParentContainer.addView(mNameContainer);
        mParentContainer.addView(mAddressContainer);
    }
}

File: Chapter06/Kotlin/CodeInterface/app/src/main/java/com/androidbook/codeinterface/MainActivity.kt
package com.androidbook.codeinterface

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private lateinit var mNameContainer: LinearLayout;
    private lateinit var mAddressContainer: LinearLayout;
    private lateinit var mParentContainer: LinearLayout;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        createNameContainer()
        createAddressContainer()
        createParentContainer()
        setContentView(mParentContainer)
    }

    private fun createNameContainer() {
        mNameContainer = LinearLayout(this)
        mNameContainer.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        mNameContainer.orientation = LinearLayout.HORIZONTAL

        val nameLbl = TextView(this)
        nameLbl.text = "Name: "

        val nameValue = TextView(this)
        nameValue.text = "John Doe"

        mNameContainer.addView(nameLbl)
        mNameContainer.addView(nameValue)
    }

    private fun createAddressContainer() {
        mAddressContainer = LinearLayout(this)
        mAddressContainer.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
        mAddressContainer.orientation = LinearLayout.HORIZONTAL

        val addrLbl = TextView(this)
        addrLbl.text = "Address:"

        val addrValue = TextView(this)
        addrValue.text = "911 Hoolywood Blvd"

        mAddressContainer.addView(addrLbl)
        mAddressContainer.addView(addrValue)
    }

    private fun createParentContainer() {
        mParentContainer = LinearLayout(this)
        mParentContainer.layoutParams = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        mParentContainer.orientation = LinearLayout.VERTICAL

        mParentContainer.addView(mNameContainer)
        mParentContainer.addView(mAddressContainer)
    }
}